cmd/containerd/command/service_windows.go

```diff
@@ -388,19 +387,14 @@ func initPanicFile(path string) error {
 
 	oldStderr = h
 
-	r, _, err := setStdHandle.Call(uintptr(sh), panicFile.Fd())
-	if r == 0 && err != nil {
-		return err
-	}
-
-	return nil
+	return windows.SetStdHandle(sh, windows.Handle(panicFile.Fd()))
 }
 
 func removePanicFile() {
 	if st, err := panicFile.Stat(); err == nil {
 		if st.Size() == 0 {
 			sh := uint32(windows.STD_ERROR_HANDLE)
-			setStdHandle.Call(uintptr(sh), uintptr(oldStderr))
+			windows.SetStdHandle(sh, oldStderr)
 			panicFile.Close()
 			os.Remove(panicFile.Name())
```

Comment:
I'm still trying to get my head around this logic;

- why do we only close `panicFile` (`panic.log`) if it's empty? (could this be the cause of some "file already in use by another process" errors in CI?
- why do we only restore StdErr if `panic.log` is empty?

If anyone has any ideas; happy to open a PR to change this

Reply:
I agree this logic seems weird. However, I didn't look through commit history for a reason why it exists this way. It does seem like we could be leaking the file handle though.

Reply:
No, not a lot of history available; most of this was added by John in https://github.com/moby/moby/commit/57aef3b49025aac6bb084491478eb461b14b9109#diff-96f1675ac02e434ec94a83543bea56ea711c396b10a1457997edf21194b6d343 (that was 7 Years ago, so I doubt he'd still know), with only a single change after that in https://github.com/moby/moby/commit/33a779e308334be55ee179087e61045c93630ec9

(heh, looks like on that one we could've caught that we could've dropped the intermediate variable to avoid the type-casting)

Reply:
I think this is ok right? We only hit this when shutting down, and we only remove when the panic file is empty. This just avoids the need on startup to move it. But ironically we check on the Open|Append that its of size 0 anyways so I think this is mostly wasted. But its not a handle leak since the process is exiting anyways. Although it would be "better" to always call close on it yes.

Reply:
Lolz